app/public/js/localSprites.js

```diff
@@ -141,6 +144,35 @@ function updateLocalSprite(localSprite) {
         // This flag will be used in the update loop to remove this sprite from the list of localSprites.
         localSprite.shouldBeRemoved = true;
     }
+    //geomtry collision checks for projectiles that die on impact
+    //BROKEN: It looks to me like "alignToTile" isn't happening, and the fireball is detonating in the frame before it would impact something,
```

Comment:
I don't see any reason this wouldn't be working, maybe the problem is that the align to tile aligns the edge of the fireball to the tile edge, but when it impacts and explodes, you are expecting the center of the explosion to be on the edge of the tile? I haven't tested this yet, so I don't actually know what it looks like. I'm just speculating at what may be happening.